### PR TITLE
Updated readme with latest changes

### PR DESCRIPTION
**Pull request by twop**
Also added the list with breaking changes

### CODE REVIEW

**Comment by twop** [comment]
I'm not native in English. So please review for any grammar mistakes as well (sry :( )

File: README.md [hunk 1]
```diff
@@ -11,15 +11,15 @@ yarn add unionize
 ## Example
 
 Call `unionize` on a record literal mapping tag literals to value types:
-```ts
+```typescript
```
Comment (lanovoy) [inline README.md]: I think 'ts' is the correct alias for typescript markup?

Comment (twop) [reply README.md]: both are fine AFAIK. But if there are any other changes I will revert it back. Thanks for noticing :)

File: README.md [hunk 2]
```diff
@@ -47,20 +47,32 @@ store.dispatch(Action.SET_VISIBILITY_FILTER('SHOW_COMPLETED'));
...
+  });
 ```
 
+ `action` can be omitted. In that case the result of match is a function
+
+```ts
+const getIdFromAction = Action.match({
+  ADD_TODO: ({ id}) => id,
+  TOGGLE_TODO: ({ id }) => id,
+  default: a => `${a.tag} doesn't have id`
```
Comment (pelotom) [inline README.md]: Just noticed, this should be `a.type`, no?

File: README.md [hunk 3]
```diff
@@ -47,20 +47,32 @@ store.dispatch(Action.SET_VISIBILITY_FILTER('SHOW_COMPLETED'));
...
+  Action.match(action, {
+    // handle cases as pure functions instead of switch statements
+    ADD_TODO: ({ id, text }) => [...state, { id, text, completed: false }],
+    TOGGLE_TODO: ({ id }) =>
+      state.map(
+        todo =>
+          todo.id === id ? { ...todo, completed: !todo.completed } : todo
+      ),
+    // handles the rest; if not provided, cases must be exhaustive 
+    default: a => state // a===action. Useful for curried version.
```
Comment (pelotom) [inline README.md]: It's a bit unfortunate now that the first example of `match` would be better implemented with `transform`.

Comment (twop) [reply README.md]: 👍 I don't know what to do about it honestly :) 
Also when I tried to put docs for transform I kept questioning myself is it enough value to justify  extra api surface. I still like 70% positive on that.

Also take a look at my take on union https://github.com/twop/ts-union. `if` api is kinda somewhere in between match and transform. I'm wondering if unionize should have it instead

Comment (pelotom) [reply README.md]: I still think `transform` is and useful shorthand.

From a cursory inspection, it doesn't seem like `if` provides any more expressive power:

```ts
// typeof str === string
const str = PaymentMethod.if.Check(
  cash,
  n => `check num: ${n.toString()}`,
  _v => 'not check' // _v is the union obj that is passed in
); // str === 'not check'
```

vs

```ts
// typeof str === string
const str = PaymentMethod.match(
  cash,
  Check: n => `check num: ${n.toString()}`,
  default: _v => 'not check' // _v is the union obj that is passed in
); // str === 'not check'
```

Am I missing something?

**Comment by pelotom** [comment]
Continued this at #26.

**Pull request merged**